examples/solvers/find_optimal_parameters.py

```diff
@@ -0,0 +1,189 @@
+if reconstruction_method == 'fbp':
+    # Define the reconstruction operator for FBP reconstruction
+    def reconstruction(proj_data, lam):
+        fbp_op = odl.tomo.fbp_op(ray_trafo,
+                                 filter_type='Hann', frequency_scaling=1 / lam)
+        return fbp_op(proj_data)
+
+    initial_param = 1.0
+
+elif reconstruction_method == 'huber':
+    # Define the reconstruction operator for Huber regularized reconstruction
+    # See lbfgs_tomograhpy_tv.py for more information.
+    def reconstruction(proj_data, parameters):
+        # Extract the separate parameters
+        lam, sigma = parameters
+
+        print('lam = {}, sigma = {}'.format(lam, sigma))
+
+        # We do not allow negative paramters, so return a bogus result
+        if lam <= 0 or sigma <= 0:
+            return 1000 * space.one()
+
+        # Create data term ||Ax - b||_2^2
+        l2_norm = odl.solvers.L2NormSquared(ray_trafo.range)
+        data_discrepancy = l2_norm * (ray_trafo - proj_data)
+
+        # Create regularizing functional huber(|grad(x)|)
+        gradient = odl.Gradient(space)
+        l1_norm = odl.solvers.GroupL1Norm(gradient.range)
+        smoothed_l1 = odl.solvers.MoreauEnvelope(l1_norm, sigma=sigma)
+        regularizer = smoothed_l1 * gradient
+
+        # Create full objective functional
+        obj_fun = data_discrepancy + lam * regularizer
+
+        # Create initial estimate of the inverse Hessian by a diagonal estimate
+        opnorm = odl.power_method_opnorm(ray_trafo)
+        hessinv_estimate = odl.ScalingOperator(space, 1 / opnorm ** 2)
+
+        # Pick parameters
+        maxiter = 30
+        num_store = 5
+
+        # Run the algorithm
+        x = ray_trafo.domain.zero()
+        odl.solvers.bfgs_method(
+            obj_fun, x, maxiter=maxiter, num_store=num_store,
+            hessinv_estimate=hessinv_estimate)
+
+        return x
+
+    initial_param = [0.1, 0.05]
+
+elif reconstruction_method == 'tv':
+    # Define the reconstruction operator for TV regularized reconstruction
+    # See chambolle_pock_tomography.py for more information.
+
+    def reconstruction(proj_data, lam):
+        lam = float(lam)
+
+        print('lam = {}'.format(lam))
+
+        # We do not allow negative paramters, so return a bogus result
+        if lam <= 0:
+            return 1000 * space.one()
+
+        # Construct operators and functionals
+        gradient = odl.Gradient(space)
+        op = odl.BroadcastOperator(ray_trafo, gradient)
+
+        g = odl.solvers.ZeroFunctional(op.domain)
+
+        l2_norm = odl.solvers.L2NormSquared(ray_trafo.range).translated(proj_data)
+        l1_norm = lam * odl.solvers.GroupL1Norm(gradient.range)
+        f = odl.solvers.SeparableSum(l2_norm, l1_norm)
+
+        # Select solver parameters
+        op_norm = 1.1 * odl.power_method_opnorm(op)
+
+        niter = 200  # Number of iterations
+        tau = 1.0 / op_norm  # Step size for the primal variable
```

Comment:
Remove "# Step size for the primal variable" style parameters, move these into the solver call